numpy/core/src/multiarray/mapping.c

```diff
@@ -139,6 +139,150 @@ PyArray_MapIterSwapAxes(PyArrayMapIterObject *mit, PyArrayObject **ret, int getm
     *ret = (PyArrayObject *)new;
 }
 
+/**
+ * Prepare an index argument into a c-array of indices.
+ *
+ * Returns the number of indices, or -1 on failure
+ *
+ * This mainly implements the following section from the advanced indexing docs:
+ * > In order to remain backward compatible with a common usage in Numeric,
+ * > basic slicing is also initiated if the selection object is any non-ndarray
+ * > sequence (such as a list) containing slice objects, the Ellipsis object,
+ * > or the newaxis object, but not for integer arrays or other embedded
+ * > sequences.
+ *
+ * This promotes scalars to 1-tuples.
+ *
+ * @param the index object, which may or may not be a tuple
+ *
+ * @returns the index converted to null-terminated pyobject array. The caller
```

Comment:
Yep, careless leftover from an old attempt that dynamically allocated memory. Updated with better comments